api/services/S3Helper.js

```diff
@@ -102,6 +108,10 @@ class S3Client {
     let wereS3ResultsTruncated = false;
     // Concatenate S3 pages until there are enough for OUR page size
     for await (const page of paginator) {
+      if (!page.Contents) {
```

Comment:
@drew in response to your [question](https://github.com/cloud-gov/pages-core/pull/4725#pullrequestreview-2638229426), we were throwing the error when an S3 path doesn't have any contents in the path so I added this check to return an empty array.